src/Components/Router.cpp

```diff
@@ -0,0 +1,124 @@
+#include "Controller.h"
+#include "Decoder.h"
+#include "Router.h"
+
+Router::Router(Decoder* decoder_,Controller* controller_) {
+    decoder = decoder_;
+    controller = controller_;
+    living = true;
+    WSAData wsaData;
+    if (WSAStartup(MAKEWORD(2, 2), &wsaData)) {
```

Comment:
For windows specific code, please enclose it within
#ifdef
#endif

Reply:
Usually we do the library startup code in main() so tha they are not tied to a particular class instance.

Reply:
Consider using library for socket to make unit testing easier.